src/transmission_manager.rs

```diff
@@ -0,0 +1,293 @@
+    // schedule the infection attempt: this grabs a contact at the time of the infection event
+    // to make sure the contacts are based on who is alive at that time and evaluates whether the
+    // transmission event is successful
+    context.add_plan(
+        context.get_current_time() + next_infection_time_gi,
+        move |context| {
+            infection_attempt(
+                context,
+                transmitee_id,
+                gi,
+                num_infection_attempts_remaining,
+                next_infection_time_unif,
+            )
+            .expect("Error finding contact in infection attempt");
+        },
+    );
+}
+
+fn handle_infectious_status_change(
+    context: &mut Context,
+    event: PersonPropertyChangeEvent<InfectiousStatusType>,
+    r_0: f64,
+    gi: f64,
+) {
+    // is the person going from S --> I?
+    // we don't care about the other cases here
+    if matches!(event.previous, InfectiousStatus::Susceptible) {
+        // get the number of infection attempts this person will have
+        // ok to use unwrap here because we pass inputs (r_0) through a validator
+        let num_infection_attempts =
+            context.sample_distr(TransmissionRng, Poisson::new(r_0).unwrap());
+        // start scheduling infection attempt events for this person
+        // people who have num_infection_attempts = 0 are still passed through this logic so that
+        // they are set to recovered
+        schedule_next_infection_attempt(context, event.person_id, gi, num_infection_attempts, 0.0);
+    }
+}
+
+pub fn init(context: &mut Context) {
+    let parameters = context
+        .get_global_property_value(Parameters)
+        .unwrap()
+        .clone();
+    context.subscribe_to_event(
+        move |context, event: PersonPropertyChangeEvent<InfectiousStatusType>| {
+            handle_infectious_status_change(
+                context,
+                event,
+                parameters.r_0,
+                parameters.generation_interval,
+            );
+        },
+    );
+}
+
+#[cfg(test)]
+mod test {
+    use crate::{parameters::ParametersValues, population_loader::Alive};
+
+    use super::*;
+    use ixa::{context::Context, random::ContextRandomExt};
+
+    fn set_up(r_0: f64) -> Context {
+        let p_values = ParametersValues {
+            max_time: 10.0,
+            seed: 42,
+            r_0,
+            infection_duration: 0.1,
+            generation_interval: 3.0,
+            report_period: 1.0,
+            synth_population_file: ".".to_string(),
+        };
+        let mut context = Context::new();
+        context.init_random(p_values.seed);
+        context.set_global_property_value(Parameters, p_values);
+        context
+    }
+
+    #[test]
+    fn test_transitions() {
+        // set a super small r_0 so that the person has a very low probability of infecting others
+        let mut context = set_up(0.000_000_000_000_000_01);
```

Comment:
Why does this matter? You only have one person in the simulation, so even if you were definitely going to infect someone, the simulation should terminate the same way

Reply:
Thanks. Currently, the skeleton of the `get_contact` method in this PR raises an error when trying to query a contact in a population of one person. So, if we set $r_0 = 1$, the agent would query a contact, and an error would be raised.

I implemented the `get_contact` this way because to me trying to query a contact in a population of one means something is wrong with your simulation set up and should raise an error. On the other hand, being unable to find an eligible contact should return `None`, and I wanted a way to differentiate between the two. However, for now we can certainly group these two together while we are just trying to get a skeleton in place.